experimental/ModelBuilder/BUILD

```diff
@@ -23,6 +23,7 @@ cc_library(
     hdrs = ["ModelBuilder.h"],
     deps = [
         "@llvm-project//mlir:Affine",
+        "@llvm-project//mlir:AllPassesAndDialectsNoRegistration",
```

Comment:
Are you sure you want to pull in all those deps? For IREE I created a target `init_mlir_passes_and_dialects` to pull in a reduced number of deps, see:
https://github.com/google/iree/blob/6b7b71bdcf024caf8b302b06a9ee608f9126a997/iree/tools/BUILD#L97-L98

Reply:
I’m trying to avoid dependency between ModelBuilder common library and the rest of iree. We are considering moving ModelBuilder in MLIR core at some point so avoid dependency is important. Note that some tests do depend on iree for some passes but ideally the core library should be independent so that it can be moved in mlir core if needed.